spec/requests/api/fikas/similarity_spec.rb

```diff
@@ -0,0 +1,47 @@
+describe 'POST and GET /api/fikas, similarity functionality', type: :request do
+  subject { response }
+  let(:admin) { create(:admin) }
+  let(:credentials) { admin.create_new_auth_token }
+
+  describe 'when two participants have identical properties' do
+    let!(:participant_1) do
+      create(:participant, name: 'Fraser', start_date: Time.now - 2.years, management: false, department: 'HR')
+    end
+    let!(:participant_2) do
+      create(:participant, name: 'Max', start_date: Time.now - 2.years, management: false, department: 'HR')
+    end
+
+    before do
+      post '/api/fikas',
+           headers: credentials
+      get '/api/fikas'
+    end
+
+    it 'is expected to return 1 possible fika for 2 participants' do
+      expect(response_json['fikas'].count).to eq 1
+    end
+
+    it 'is expected to return a similarity score that is 1' do
+      expect(response_json['fikas'].last['similarity']).to be_within(0.00001).of(1.0)
```

Comment:
As this is expected to be 1, should we take away the to be_within?

Reply:
It will never be exactly one due to floating point arithmetic shortcomings. See below for why:
https://en.wikipedia.org/wiki/Floating-point_arithmetic
https://floating-point-gui.de/
I found the `be_within` together with such a small variance to be an acceptable workaround, I'll change the language on 24 to clarify what's going on.